Sources/Danger/GitLabDSL.swift

```diff
@@ -237,3 +278,30 @@ extension GitLab {
         public let webUrl: String
     }
 }
+
+// MARK: Custom decoder for Milestone
+
+extension GitLab.MergeRequest.Milestone {
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        if let id = try container.decodeIfPresent(Int.self, forKey: .groupId) {
+            parent = .group(id)
+        } else if let id = try container.decodeIfPresent(Int.self, forKey: .projectId) {
+            parent = .project(id)
+        } else {
+            throw ParentIdentifier.Error.validKeyNotFound
+        }
+
+        createdAt = try container.decode(Date.self, forKey: .createdAt)
+        description = try container.decode(String.self, forKey: .description)
+        dueDate = try container.decode(Date?.self, forKey: .dueDate)
+        id = try container.decode(Int.self, forKey: .id)
+        iid = try container.decode(Int.self, forKey: .iid)
+        startDate = try container.decode(Date?.self, forKey: .startDate)
+        state = try container.decode(State.self, forKey: .state)
+        title = try container.decode(String.self, forKey: .title)
+        updatedAt = try container.decode(Date.self, forKey: .updatedAt)
+        webUrl = try container.decode(String.self, forKey: .webUrl)
+    }
+}
```

Comment:
To prevent the lost of generated initializer, in the end, I added the custom decoder over here.

Reply:
It is kind of sacrifice for me due to after I added the processing of parent identifier I have to write all the decoding statements out. 

I have tried the advice you gave earlier but it not works out cause we are not able to point `group_id` and `project_id` to the same coding key at the same time. So that I have to raise the decoding to the upper level.

Please let me know if there are better solutions for this kind of situation. Thanks :)